core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -1111,6 +1111,10 @@ class ZKCheckedEphemeral(path: String,
                             case Code.INVALIDACL =>
                               error("Invalid ACL")
                               setResult(Code.INVALIDACL)
+                            case Code.NOAUTH =>
+                              // Allows old consumer to register with secure ZK
+                              error("NOAUTH raised on path %s (AUTH may not be needed to create children)".format(path))
+                              createRecursive(path, suffix)
```

Comment:
Hmm.. Do we really want to try again? We'd most likely just see NOAUTH again, right? I thought what we discussed offline is to check whether the node exists and if so, just return.